packages/@lwc/engine-core/src/libs/aria-reflection/aria-reflection.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import {
+    AriaPropNameToAttrNameMap,
+    defineProperty,
+    getOwnPropertyDescriptor,
+    isNull,
+    isUndefined,
+    keys,
+} from '@lwc/shared';
+
+// Apply ARIA string reflection behavior to a prototype.
+// This is deliberately kept separate from @lwc/aria-reflection. @lwc/aria-reflection is a global polyfill that is
+// needed for backwards compatibility in LEX, whereas `applyAriaReflection` is designed to only apply to our own
+// LightningElement/BaseBridgeElement prototypes.
+export function applyAriaReflection(prototype: any) {
+    for (const propName of keys(AriaPropNameToAttrNameMap)) {
+        const attrName = AriaPropNameToAttrNameMap[propName];
+        if (isUndefined(getOwnPropertyDescriptor(prototype, propName))) {
+            // Note that we need to call this.{get,set,has,remove}Attribute rather than dereferencing
+            // from Element.prototype, because these methods are overridden in LightningElement.
+            defineProperty(prototype, propName, {
+                get(this: HTMLElement): any {
+                    return this.getAttribute(attrName);
```

Comment:
Correct, everything is strings – including prop keys and values. This is actually what Chrome/Safari does today, although it is the source of a lot of disagreement over the spec because some folks think that e.g. `aria-posinset` should really be a number rather than a string. AIUI this is the main reason [Firefox has not implemented yet](https://bugzilla.mozilla.org/show_bug.cgi?id=1785412).

In any case, our historical behavior here is to align with current Safari/Chrome and treat everything as strings.